pkg/manager/member/tikv_scaler.go

```diff
@@ -67,6 +72,33 @@ func (tsd *tikvScaler) ScaleOut(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulS
 		return err
 	}
 
+	tcName := tc.Name
+	namespace := tc.Namespace
+	scaleOutTikvAddresss := fmt.Sprintf("%s.%s.%s.svc", util.GetPodName(tc, v1alpha1.TiKVMemberType, ordinal), util.GetPeerServiceName(tc, v1alpha1.TiKVMemberType), namespace)
+	pdClient := tsd.pdControl.GetPDClient(pdapi.Namespace(namespace), tcName, tc.IsTLSClusterEnabled())
+	storesInfo, err := pdClient.GetStores()
+	if err != nil {
+		klog.Errorf(err.Error())
+		return err
+	}
+	// check whether there existed store which have occupied the address before scale-out. If existed, we would delete the store first
+	// and wait the next round.
+	// The existed store which have occupied the address could be caused by the tikv which start failed due to the unexpected terminal
+	// In that case, the address might been registered but the data couldn't be persisted which would occupied the address until
+	// be deleted manually.
+	for _, store := range storesInfo.Stores {
+		if strings.HasPrefix(store.Store.Address, scaleOutTikvAddresss) {
+			err := pdClient.DeleteStore(store.Store.Id)
```

Comment:
No need to delete store for each sync, you can refer to the logic in `scaleIn`.

Reply:
I don't understand the problem we are trying to solve. Before `setReplicasAndDeleteSlots`, the new tikv pod will not be created, so in which case the address will be occupied?